### PR TITLE
net/ntopng: new datadir

### PR DESCRIPTION
**Pull request by mimugmail**
datadir moved to new direction

https://forum.opnsense.org/index.php?topic=10922.0

### CODE REVIEW

File: net/ntopng/pkg-descr [hunk 1]
```diff
@@ -4,4 +4,21 @@ is based on libpcap and it has been written in a portable
...
+
+* Changed data directory to avoid warning
+
+1.1
+
+* Add https support
+
+1.0
+
+* Initial release
```
Comment (fichtner) [inline net/ntopng/pkg-descr]: Just a thought I have a few times now: "initial release" is certainly true, but giving a list of initial version features is also nice. What do you think?

File: net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf [hunk 2]
```diff
@@ -12,4 +12,5 @@
 {%   if helpers.exists('OPNsense.ntopng.general.dnsmode') and OPNsense.ntopng.general.dnsmode != '' %}
 -n={{ OPNsense.ntopng.general.dnsmode }}
 {%   endif %}
+-d=/var/db/ntopng
```
Comment (fichtner) [inline net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf]: Shouldn't this be made persistent under /var MFS? We don't seem to do this yet.

File: net/ntopng/pkg-descr (hunk 1 above)
Comment (mimugmail) [reply net/ntopng/pkg-descr]: Good idea, I'll try to keep it in mind for the next updates

File: net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf (hunk 2 above)
Comment (mimugmail) [reply net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf]: Reference code please. :) Never worked with mfs

File: net/ntopng/pkg-descr (hunk 1 above)
Comment (fichtner) [reply net/ntopng/pkg-descr]: thanks 👍

File: net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf (hunk 2 above)
Comment (fichtner) [reply net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf]: https://github.com/opnsense/plugins/blob/master/net/zerotier/src/opnsense/service/templates/OPNsense/zerotier/zerotier#L9

Comment (fichtner) [reply net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf]: Might be a good time to introduce this as the directory changes anyway. Also another bug fixed for the changelog :)

File: net/ntopng/src/opnsense/scripts/OPNsense/Ntopng/setup.sh [hunk 3]
```diff
@@ -4,8 +4,13 @@ mkdir -p /var/run/ntopng/
...
-mkdir -p /var/tmp/ntopng/
-chmod 755 /var/tmp/ntopng
-chown ntopng:wheel /var/tmp/ntopng
+mkdir -p /var/db/ntopng/
+chmod 755 /var/db/ntopng
+chown ntopng:wheel /var/db/ntopng
+
+if [ -d "/var/tmp/ntopng/" ]; then
+  mv /var/tmp/ntopng/* /var/db/ntopng/
+  rm -rf /var/tmp/ntopng/
```
Comment (fichtner) [inline net/ntopng/src/opnsense/scripts/OPNsense/Ntopng/setup.sh]:
```suggestion
  rm -rf /var/tmp/ntopng
```

File: net/ntopng/src/opnsense/scripts/OPNsense/Ntopng/setup.sh [hunk 4]
```diff
@@ -4,8 +4,13 @@ mkdir -p /var/run/ntopng/
...
 chown ntopng:ntopng /var/run/ntopng
 
-mkdir -p /var/tmp/ntopng/
-chmod 755 /var/tmp/ntopng
-chown ntopng:wheel /var/tmp/ntopng
+mkdir -p /var/db/ntopng/
+chmod 755 /var/db/ntopng
+chown ntopng:wheel /var/db/ntopng
+
+if [ -d "/var/tmp/ntopng/" ]; then
```
Comment (fichtner) [inline net/ntopng/src/opnsense/scripts/OPNsense/Ntopng/setup.sh]:
```suggestion
if [ -d /var/tmp/ntopng ]; then
```

**Comment by mimugmail** [comment]
Thanks for assistance as usual :)

**Comment by fichtner** [comment]
Merged, thanks!

File: net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf (hunk 2 above)
Comment (mimugmail) [reply net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf]: I have no idea why it's outside of if/else but you might know best :)

**Pull request merged**